### ISSUE TITLE
Conversions from `Nullable{T}` are not implemented

### ISSUE THREAD

**Issue by jakubmisek** [peachpiecompiler/peachpie#938]
ref https://github.com/peachpiecompiler/peachpie/issues/929

this is result of https://github.com/peachpiecompiler/peachpie/issues/934, the following must be fixed

- [x] echo `Nullable` results
- [x] isset() does not respect nullable values
- [x] conversions in general does not respect nullable value (should be ~ `HasValue ? Convert(Value) : Convert(NULL)`)
- [x] Nullable to double https://github.com/peachpiecompiler/peachpie/issues/929#issuecomment-819237083
- [x] ?bool to string results in "True", "False" - should be "1", ""
- [x] conversion from nullable in runtime
- [ ] EnsureArray( Nullable{T} )

**Comment by menturion** [comment]
@jakubmisek 

Just tested. I'm getting the following issue 

`.nuget\packages\peachpie.net.sdk\1.0.0-appv4629\build\Peachpie.NET.Core.Sdk.targets(245,5): error : EnsureArray(Nullable) not implemented at \vendor\paragonie\sodium_compat\src\Compat.php(3260, 21)`

which corresponds to this line of code 
https://github.com/paragonie/sodium_compat/blob/efcecfbceac1bf014b1ede38f4588931fef92586/src/Compat.php#L3363

**Comment by jakubmisek** [comment]
now the `Compat.php` should compile

**Comment by menturion** [comment]
@jakubmisek 

Now I'm getting an `Object reference not set to an instance of an object` issue, see...

```
  PeachPie PHP Compiler version 1.0.0-appv4630+d25cb3e65ecac80542d2303807f6d0ce0f264c37
.nuget\packages\peachpie.net.sdk\1.0.0-appv4630\build\Peachpie.NET.Core.Sdk.targets(245,5): error : Object reference not set to an instance of an object. [projectname.msbuildproj]

at Pchp.CodeAnalysis.CodeGen.CodeGenerator.CanBeNull(TypeRefMask tmask) 
at Pchp.CodeAnalysis.CodeGen.CodeGenerator.EmitConvertToNullable_T(TypeSymbol from, TypeRefMask fromHint, TypeSymbol nullabletype, TypeSymbol ttype, ConversionKind conversion)
at Pchp.CodeAnalysis.CodeGen.CodeGenerator.EmitConvert(TypeSymbol from, TypeRefMask fromHint, TypeSymbol to, ConversionKind conversion)
at Pchp.CodeAnalysis.CodeGen.CodeGenerator.EmitCall(ILOpCode code, MethodSymbol method, BoundExpression thisExpr, ImmutableArray`1 arguments, BoundTypeRef staticType)
at Pchp.CodeAnalysis.CodeGen.CodeGenerator.EmitForwardCall(MethodSymbol target, MethodSymbol thismethod, IPlace thisPlaceExplicit, Boolean callvirt)
at Pchp.CodeAnalysis.CodeGen.GhostMethodBuilder.<>c__DisplayClass2_0.<GenerateGhostBody>b__0(ILBuilder il)
at Pchp.CodeAnalysis.CodeGen.MethodGenerator.GenerateMethodBody(PEModuleBuilder moduleBuilder, MethodSymbol routine, Action`1 builder, VariableSlotAllocator variableSlotAllocatorOpt, DiagnosticBag diagnostics, Boolean emittingPdb)
at Pchp.CodeAnalysis.CodeGen.GhostMethodBuilder.GenerateGhostBody(PEModuleBuilder module, DiagnosticBag diagnostic, MethodSymbol method, SynthesizedMethodSymbol ghost)
at Pchp.CodeAnalysis.CodeGen.GhostMethodBuilder.CreateGhostOverload(MethodSymbol method, NamedTypeSymbol containingtype, PEModuleBuilder module, DiagnosticBag diagnostic, TypeSymbol ghostreturn, ImmutableArray`1 ghostparams, Boolean phphidden, MethodSymbol explicitOverride)
at Pchp.CodeAnalysis.Symbols.SourceTypeSymbol.FinalizeMethodTable(PEModuleBuilder module, DiagnosticBag diagnostics)
at Pchp.CodeAnalysis.EnumeratorExtension.ForEach[T](IEnumerable`1 enumerable, Action`1 func)
at Pchp.CodeAnalysis.SourceCompiler.WalkTypes(Action`1 action, Boolean allowParallel)
at Pchp.CodeAnalysis.SourceCompiler.EmitSynthesized()
at Pchp.CodeAnalysis.SourceCompiler.CompileSources(PhpCompilation compilation, PEModuleBuilder moduleBuilder, Boolean emittingPdb, Boolean hasDeclarationErrors, DiagnosticBag diagnostics, CancellationToken cancellationToken)
at Pchp.CodeAnalysis.PhpCompilation.CompileMethods(CommonPEModuleBuilder moduleBuilder, Boolean emittingPdb, Boolean emitMetadataOnly, Boolean emitTestCoverageData, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken)
at Microsoft.CodeAnalysis.CommonCompiler.CompileAndEmit(TouchedFileLogger touchedFilesLogger, Compilation& compilation, ImmutableArray`1 analyzers, ImmutableArray`1 generators, ImmutableArray`1 additionalTextFiles, AnalyzerConfigSet analyzerConfigSet, ImmutableArray`1 sourceFileAnalyzerConfigOptions, ImmutableArray`1 embeddedTexts, DiagnosticBag diagnostics, CancellationToken cancellationToken, CancellationTokenSource& analyzerCts, Boolean& reportAnalyzer, AnalyzerDriver& analyzerDriver) in \jmise\Projects\Peachpie.Microsoft.CodeAnalysis\src\Compilers\Core\Portable\CommandLine\CommonCompiler.cs:line 1069
at Microsoft.CodeAnalysis.CommonCompiler.RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, CancellationToken cancellationToken) in \jmise\Projects\Peachpie.Microsoft.CodeAnalysis\src\Compilers\Core\Portable\CommandLine\CommonCompiler.cs:line 811
at Microsoft.CodeAnalysis.CommonCompiler.Run(TextWriter consoleOutput, CancellationToken cancellationToken) in \jmise\Projects\Peachpie.Microsoft.CodeAnalysis\src\Compilers\Core\Portable\CommandLine\CommonCompiler.cs:line 699
at Peachpie.NET.Sdk.Tools.BuildTask.Execute()
```

**Comment by jakubmisek** [comment]
thanks @menturion :) adding few more tests and if won't be able to make it reliably working, this change will be reverted

fixed in `1.0.0-appv4631`

**Comment by menturion** [comment]
The build process (of the complete project) just finished successfully w/o throwing an error. ... I swear :) 

I will do the runtime tests regarding https://github.com/peachpiecompiler/peachpie/issues/929#issuecomment-818738060 later.
